metron-deployment/vagrant/Kerberos-setup.md

```diff
@@ -87,147 +86,159 @@ sudo -u hdfs hdfs dfs -chmod 770 /user/metron
+        storeKey=true
+        useTicketCache=false
+        serviceName="kafka"
+        principal="metron@EXAMPLE.COM";
+    };
+    EOF
+    ```
 
 18. Create a storm.yaml with jaas file info. Set the array of nimbus hosts accordingly.
-  ```
-[metron@node1 .storm]$ cat storm.yaml
-nimbus.seeds : ['node1']
-java.security.auth.login.config : '/home/metron/.storm/client_jaas.conf'
-storm.thrift.transport : 'org.apache.storm.security.auth.kerberos.KerberosSaslTransportPlugin'
-  ```
+    ```
+    cat << EOF > storm.yaml
+    nimbus.seeds : ['node1']
+    java.security.auth.login.config : '/home/metron/.storm/client_jaas.conf'
+    storm.thrift.transport : 'org.apache.storm.security.auth.kerberos.KerberosSaslTransportPlugin'
+    EOF
+    ```
 
 19. Create an auxiliary storm configuration json file in the metron user’s home directory. Note the login config option in the file points to our custom client_jaas.conf.
-  ```
-cd /home/metron
-[metron@node1 ~]$ cat storm-config.json
-{
-  "topology.worker.childopts" : "-Djava.security.auth.login.config=/home/metron/.storm/client_jaas.conf"
-}
-  ```
+    ```
+    cat << EOF > ~/storm-config.json
+    {
+        "topology.worker.childopts" : "-Djava.security.auth.login.config=/home/metron/.storm/client_jaas.conf"
+    }
+    EOF
+    ```
 
 20. Setup enrichment and indexing.
 
-    a. Modify enrichment.properties - `${METRON_HOME}/config/enrichment.properties`
-
+    a. Modify enrichment.properties as root located at `${METRON_HOME}/config/enrichment.properties`
     ```
-    kafka.security.protocol=PLAINTEXTSASL
-    topology.worker.childopts=-Djava.security.auth.login.config=/home/metron/.storm/client_jaas.conf
+    if [[ $EUID -ne 0 ]]; then
+        echo -e "\nERROR:\tYou must be root to run these commands.  You may need to type exit."
+    else
+        sed -i 's/kafka.security.protocol=.*/kafka.security.protocol=PLAINTEXTSASL/' ${METRON_HOME}/config/enrichment.properties
+        sed -i 's/topology.worker.childopts=.*/topology.worker.childopts=-Djava.security.auth.login.config=\/home\/metron\/.storm\/client_jaas.conf/' ${METRON_HOME}/config/enrichment.properties
+    fi
     ```
 
-    b. Modify elasticsearch.properties - `${METRON_HOME}/config/elasticsearch.properties`
-
+    b. Modify elasticsearch.properties as root located at `${METRON_HOME}/config/elasticsearch.properties`
     ```
-    kafka.security.protocol=PLAINTEXTSASL
-    topology.worker.childopts=-Djava.security.auth.login.config=/home/metron/.storm/client_jaas.conf
+    if [[ $EUID -ne 0 ]]; then
+        echo -e "\nERROR:\tYou must be root to run these commands.  You may need to type exit."
+    else
+        sed -i 's/kafka.security.protocol=.*/kafka.security.protocol=PLAINTEXTSASL/' ${METRON_HOME}/config/elasticsearch.properties
+        sed -i 's/topology.worker.childopts=.*/topology.worker.childopts=-Djava.security.auth.login.config=\/home\/metron\/.storm\/client_jaas.conf/' ${METRON_HOME}/config/elasticsearch.properties
+    fi
```

Comment:
I kind of prefer just showing the key=value so that it is clear to the user what properties need to change.

I appreciate the fact that you've automated it, but since these are manual instructions, maybe we should opt for making them as clear as possible. I could be convinced otherwise though.